src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/DamageableItem.java

```diff
@@ -69,4 +81,26 @@ default void damageItem(@Nonnull Player p, @Nullable ItemStack item) {
         }
     }
 
+    /**
+     * This method will randomly decide if the item should be damaged or not
+     * This does not damage the item, it is called by {@link #damageItem(Player, ItemStack)} to randomly generate a boolean
+     *
+     * @param unbreakingLevel
+     *                                 The {@link Integer} level of the unbreaking enchantment
+     * @throws IllegalArgumentException
+     *                                 Thrown when {@link #itemType()} doesn't return a valid {@link UnbreakingAlgorithms} value
+     * @return Whether you should keep the item undamaged
+     *
+     */
+    default boolean evaluateUnbreakingEnchantment(int unbreakingLevel) {
+        switch (itemType()){
+            case TOOLS:
+                return !(Math.random() < (1.0 / (unbreakingLevel + 1)));
+            case ARMORS:
+                return !(Math.random() < 0.6 + (0.4 / (unbreakingLevel + 1)));
+            default:
+                throw new IllegalArgumentException("itemType must be either UnbreakingAlgorithms.TOOLS or UnbreakingAlgorithms.ARMORS");
+        }
```

Comment:
This is dead code, an enum is used to ensure that there are only TOOLS or ARMOR as possible types.
A null value would throw a NullPointerException, so this code is never reached.

Reply:
My IDE would raise an error if I didn't put that

Reply:
Defaults should always be there. People can still technically inject into enums, new things can be aded, etc. It's good practice to include it.

SonarLint must complain if it doesn't have it, no?